### PR TITLE
Ready for Review - Updating sensu

### PR DESCRIPTION
**Pull request by genagain**
In this pull request, I added a threshold flag on the cpu-pcnt-usage-metrics, disk-usage-metrics, and memory-metrics plugins, so it can trigger an email alert if it is surpassed. I also added the instructions retired plugin to count the number of retired instructions per second.

### CODE REVIEW

**Comment by genagain** [comment]
I ran puppet agent -t on both nodes 25 and 46 the their sensu configurations are just like I expected!

**Comment by genagain** [comment]
I ran `puppet agent -t` on both nodes 25 and 46 the their sensu configurations are just like I expected!

File: sensu/files/plugins/check-ip-connectivity.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ STATE_DEPENDENT=4
 
 #target=
 
-count=$(ping -c 1 10.13.37.141 | grep icmp* | wc -l)
+count=$(ping -c 1 129.10.3.55 | grep -e "icmp_seq=1 ttl=*" | wc -l)
```
Comment (rahulait) [inline sensu/files/plugins/check-ip-connectivity.sh:14]: let's remove the ip-address and add variable here.

**Comment by genagain** [comment]
This works on both my compute and controller vms in staging!

**Comment by rahulait** [comment]
From where is check-ip-connectivity.sh getting called? Is it also passing the parameter? Can you please point to that code.

**Comment by genagain** [comment]
The check-ip-connectivity.sh is getting called on the sensu server, where it is configured on the Sensu server. An example of an configuration is below.

```
{
  "checks": {
    "ip_check": {
      "handlers": ["default"],
      "command": "/etc/sensu/plugins/check-ip-connectivity.sh 10.13.37.98",
      "interval": 60,
      "subscribers": [ "moc-sensu" ]
    }
  }
}
```

**Comment by rahulait** [comment]
+1 if its working fine on node-25.

**Comment by genagain** [comment]
I tested it on node-25 before the foreman on Friday and it worked. Because I don't have an account on the foreman vm node 16 I cannot test it again on node 25 until I have the account.

**Comment by genagain** [comment]
Thanks for the +1 @rahulait!

**Comment by genagain** [comment]
Jon and I tested it on 25 again and it worked.

**Comment by genagain** [comment]
I removed the check that required sensu to have sudo access. Now that sensu doesn't need sudo access we don't need an `/etc/sudoers.d/sensu` file which was causing all of the problems with neutron.

**Comment by genagain** [comment]
I tested the updated configurations in my staging environment and they work like expected!

**Comment by rahulait** [comment]
I guess there are too many commits in this pull-request. There are too many things done again and again. Can we close this one and submit a fresh one with minimal changes?

**Comment by genagain** [comment]
I am closing with PR because this has way too many commits. A PR with the most essential commits is linked below.

https://github.com/CCI-MOC/kilo-puppet/pull/31

**Pull request closed without merging**